scanapp/views.py

```diff
@@ -161,3 +123,79 @@ def post(self, request):
                 }
             )
         )
+
+
+class UrlScanView(FormView):
+    template_name = 'scanapp/urlscan.html'
+    form_class = UrlScanForm
+
+    def post(self, request, *args, **kwargs):
+        form = self.form_class(request.POST)
+
+        if form.is_valid():
+            # get the URL from the form
+            url = request.POST['url']
+
+            logger = logging.getLogger(__name__)
+
+            if parse_url(url) == 1:
+
+                gitparser = GitURL(url)
+
+                url = gitparser.to_git()
+
+                home_path = expanduser("~")
+                os.chdir(home_path)
+
+                if (str(request.user) == 'AnonymousUser'):
+                    path = 'media/AnonymousUser/url/'
+                    user = None
+                else:
+                    path = 'media/user/' + str(request.user) + '/url/'
+                    user = request.user
+
+                scan_start_time = datetime.now()
+                subprocess.call(['mkdir', '-p', path])
+
+                file_name = ''.join(e for e in url if e.isalnum())
+
+                scan_directory = file_name
+
+                scan_id = create_scan_id(user, url, scan_directory, scan_start_time)
+                logger.info('git repo detected')
+
+                scan_code_async_final(url, scan_id, path)
+
+                # return the response as HttpResponse
+                return HttpResponseRedirect('/resultscan/' + str(scan_id))
+
+            else:
+
+                url = request.POST['url']
+
+                # different paths for both anonymous and registered users
+                if (str(request.user) == 'AnonymousUser'):
+                    path = 'media/AnonymousUser/url/'
+                    user = None
+                else:
+                    path = 'media/user/' + str(request.user) + '/url/'
```

Comment:
Also do not join path-lile arguments with +, use posixpath.join or url-related functions instead or even a simple '/'.join() would be better